File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        l=0
        r=0
        operations=0
        while(r<len(nums)):

            if(nums[r]==0):
                if r+2<len(nums):
                    for i in range(r,r+3):
                        nums[i]=1-nums[i]
                    operations+=1
            r+=1
        
        for i in range(len(nums)):
            if nums[i]==0:
                return -1
        return operations


        

File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int r=0;
        int count=0;
        while(r<nums.size()){

            if (nums[r]==0){
                if(r+2<nums.size()){
                    for(int i=r;i<r+3;i++){
                        nums[i]=1-nums[i];
                    }
                    count++;
                }
            }
            r+=1;
        }
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                return -1;
            }
        }
        return count;
    }
};